From a8f6c2256ff628971c4f736a86a997dabad92028 Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Mon, 3 Aug 2015 18:26:55 +0800
Subject: [PATCH 104/115] MA-6198 Update ProxyInfo for OMXPlayer

Update proxy information into the system property
rw.HTTP_PROXY by the format "host:port".
Then the OMX player can use this proxy information
for its own http stack.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetManager.java | 2 ++
 Ethernet/com/fsl/ethernet/MainActivity.java    | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index 3d7eaec..dbfaf71 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -434,6 +434,7 @@ public class EthernetManager {
         exclusionList=getSharedPreProxyExclusionList();
         if (getSharedPreProxyAddress() == null || getSharedPreProxyPort() == null) {
             mConnMgr.setGlobalProxy(null);
+            SystemProperties.set("rw.HTTP_PROXY", "");
             return;
         }
         LinkProperties lp = mConnMgr.getLinkProperties(ConnectivityManager.TYPE_ETHERNET);
@@ -448,6 +449,7 @@ public class EthernetManager {
             new ProxyInfo(getSharedPreProxyAddress(), port, exclusionList);
         mConnMgr.setGlobalProxy(null);
         mConnMgr.setGlobalProxy(proxyProperties);
+        SystemProperties.set("rw.HTTP_PROXY", proxyProperties.getHost() + ":" + proxyProperties.getPort());
     }
 
     public void initProxy(){
diff --git a/Ethernet/com/fsl/ethernet/MainActivity.java b/Ethernet/com/fsl/ethernet/MainActivity.java
index 1b05f7c..827c3a2 100644
--- a/Ethernet/com/fsl/ethernet/MainActivity.java
+++ b/Ethernet/com/fsl/ethernet/MainActivity.java
@@ -38,6 +38,7 @@ import android.view.Window;
 import android.view.WindowManager;
 import android.content.IntentFilter;
 import android.content.Intent;
+import android.os.SystemProperties;
 
 public class MainActivity extends Activity {
     private EthernetEnabler mEthEnabler;
@@ -64,6 +65,7 @@ public class MainActivity extends Activity {
                     if (info.getType() == ConnectivityManager.TYPE_ETHERNET) {
                         if (info.getState() == State.DISCONNECTED)
                             mConnMgr.setGlobalProxy(null);
+                            SystemProperties.set("rw.HTTP_PROXY", "");
                         if (info.getState() == State.CONNECTED)
                             mEthEnabler.getManager().initProxy();
                     }
-- 
1.8.0

